ios/FluentUI/TextField/FluentTextField.swift

```diff
@@ -0,0 +1,313 @@
+        imageTextStack.isLayoutMarginsRelativeArrangement = true
+
+        addSubview(imageTextStack)
+
+        let safeArea = safeAreaLayoutGuide
+        NSLayoutConstraint.activate([
+            separator.widthAnchor.constraint(equalTo: textStack.widthAnchor),
+            textfield.leadingAnchor.constraint(equalTo: textStack.leadingAnchor),
+            textfield.trailingAnchor.constraint(equalTo: textStack.trailingAnchor),
+            imageTextStack.leadingAnchor.constraint(equalTo: safeArea.leadingAnchor),
+            imageTextStack.trailingAnchor.constraint(equalTo: safeArea.trailingAnchor),
+            imageTextStack.topAnchor.constraint(equalTo: safeArea.topAnchor),
+            imageTextStack.bottomAnchor.constraint(equalTo: safeArea.bottomAnchor),
+            leadingImageContainerView.leadingAnchor.constraint(equalTo: leadingImageView.leadingAnchor),
+            leadingImageContainerView.trailingAnchor.constraint(equalTo: leadingImageView.trailingAnchor),
+            leadingImageView.centerYAnchor.constraint(equalTo: textfield.centerYAnchor)
+        ])
+
+        NotificationCenter.default.addObserver(self,
+                                               selector: #selector(themeDidChange),
+                                               name: .didChangeTheme,
+                                               object: nil)
+        updateTokenizedValues()
+
+        tokenSetSink = tokenSet.sinkChanges { [weak self] in
+            self?.updateTokenizedValues()
+        }
+    }
+
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    // MARK: UITextFieldDelegate
+    public func textFieldDidBeginEditing(_ textField: UITextField) {
+        if let onDidBeginEditing = onDidBeginEditing {
+            onDidBeginEditing(self)
+        }
+        updateState()
+    }
+
+    public func textFieldDidEndEditing(_ textField: UITextField) {
+        if let onDidEndEditing = onDidEndEditing {
+            onDidEndEditing(self)
+        }
+        updateState()
+        if text?.isEmpty == true {
+            textfield.rightViewMode = .whileEditing
+        } else {
+            textfield.rightViewMode = .always
+        }
+    }
+
+    public func textFieldShouldReturn(_ textField: UITextField) -> Bool {
+        // add check for error == nil?
+        if let onReturn = onReturn {
+            return onReturn(self)
+        }
+        // what is a good default?
+        return true
+    }
+
+    var attributedPlaceholder: NSAttributedString? {
+        guard let placeholder = placeholder else {
+            return nil
+        }
+        return NSAttributedString(string: placeholder, attributes: [.foregroundColor: UIColor(dynamicColor: tokenSet[.placeholderColor].dynamicColor)])
+    }
+
+    // The leadingImageView needs a contianer to be vertically centered on the
+    // textfield
+    let leadingImageContainerView: UIView = {
```

Comment:
i thought simple variables, sometimes private lazy var is more binary efficient?

Reply:
Looks like the lib.a size goes down when you switch to lazy var, but the demo app goes up. So I think I'll keep the lets for now.
Before
 	lib.a - 29,524,616 bytes 
	app - 32,213,016 bytes
	app.app unstripped - 20,537,024 bytes
	app.app stripped - 7,472,080 bytes
After 
	lib.a - 29,523,296 bytes
	app - 32,215,064 bytes
	app.app unstripped - 20,539,072 bytes
	app.app stripped - 7,472,784 bytes